targets/drupal7/Drupal7Target.php

```diff
@@ -31,6 +31,15 @@ public function install(): void {
       );
     }
 
+    copy('compress.zlib://' . __DIR__.'/settings.php.gz', __DIR__.'/temp.php');
```

Comment:
This section is modifying the files that get checked in.  We don't want to do that if possible.  This should modify the files after they are copied to the tmp directory used for the test.

Also consider putting a pattern in the settings.php file that you match for.  Something like:
`__DB_HOST__`.  You can find other parts of this project that use that style.